### PR TITLE
feat(contentpreview): Make ContentPreview location aware

### PR DESCRIPTION
**Pull request by ConradJChan**

### CODE REVIEW

File: src/elements/common/nav-router/NavRouter.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import * as React from 'react';
+import { MemoryRouter, MemoryRouterProps, Router } from 'react-router';
+import { History } from 'history';
+
+type Props = {
+    children: React.ReactNode;
+    history?: History;
+} & MemoryRouterProps;
```
Comment (jstoffan) [inline src/elements/common/nav-router/NavRouter.tsx:8]: This isn't present on the current component. Will it require developers to pass in a router and/or its props? It's meant to be an optional feature. Will it work with non-memory routers?

File: src/elements/common/nav-router/index.js [hunk 2]
```diff
@@ -1 +1,2 @@
 export { default } from './NavRouter';
```
Comment (jstoffan) [inline src/elements/common/nav-router/index.js:1]: Should we remove NavRouter.js or convert it to a `.js.flow` file?

File: src/elements/content-preview/__tests__/PreviewNavigation.test.js [hunk 3]
```diff
@@ -48,16 +51,18 @@ describe('elements/content-preview/PreviewNavigation', () => {
...
-                    intl={{
-                        formatMessage: jest.fn(),
-                    }}
-                    collection={['a', 'b', 'c']}
-                    currentIndex={2}
-                    onNavigateLeft={onNavigateLeftMock}
-                    onNavigateRight={jest.fn()}
-                    history={historyMock}
-                />,
+                <Router history={historyMock}>
```
Comment (jstoffan) [inline src/elements/content-preview/__tests__/PreviewNavigation.test.js:54]: Should this use `getWrapper`?

File: src/elements/common/nav-router/NavRouter.tsx (hunk 1 above)
Comment (ConradJChan) [reply src/elements/common/nav-router/NavRouter.tsx:8]: Yeah I think due to flow, it’s not restricted but ContentSidebar makes use of the “hidden” props for the MemoryRouter. They’re optional anyway so users don’t have to pass anything in. I think this is just a more accurate typing

File: src/elements/content-preview/__tests__/PreviewNavigation.test.js (hunk 3 above)
Comment (ConradJChan) [reply src/elements/content-preview/__tests__/PreviewNavigation.test.js:54]: Yeah I’ll try to clean this up

File: src/elements/common/nav-router/NavRouter.tsx (hunk 1 above)
Comment (jstoffan) [reply src/elements/common/nav-router/NavRouter.tsx:8]: I see. Does this work with non-memory routers? If I pass in a BrowserRouter, will Typescript complain?

**Pull request merged**